server/filestore.go

```diff
@@ -1293,10 +1293,16 @@ func (mb *msgBlock) convertCipher() error {
 
 		buf, _ := mb.loadBlock(nil)
 		bek.XORKeyStream(buf, buf)
+		// Check for compression.
+		nbuf := buf
+		if nbuf, err = mb.decompressIfNeeded(nbuf); err != nil {
```

Comment:
Might be. clearer is arg is `buf` and nbuf is `var nbuf []byte`

Reply:
Have updated to:
```go
// Check for compression, and make sure we can parse with old cipher and key file.
if nbuf, err := mb.decompressIfNeeded(buf); err != nil {
	return err
} else if err = mb.indexCacheBuf(nbuf); err != nil {
	return err
}
```